### ISSUE TITLE
Task runner daemon

### ISSUE THREAD

**Issue by nikosdion** [akeeba/panopticon#332]
CRON jobs are great, but as you are handling more and more sites there comes a point where you have a ridiculous amount of them, even though they mostly sit on their butt doing nothing.

The solution to that is having a _daemon_ (service) which is launched at system startup and checks for new work every couple of seconds by _forking_ a process. Obviously, this only works with Linux servers (maybe with macOS too?) as it requires the POSIX and `pcntl` extensions for PHP.

The best chance at doing this is using something like `https://github.com/lifo101/php-daemon/tree/master/src`.

We would only need on daemon Task which runs our own task runner command once.

Parameters:
* `loopInterval`: 1.0
* `loopSleepMin`: 10000
* `autoRestartInterval`: the value of `max_execution`
* `minRestartThreshold`: 10
* `daemonize`: true

Before launching a new task we should consider the following:
* Do we already have the maximum number of tasks executing? This should be configurable with the default set to 10
* Have we hit the minimum allowed free memory? Ignore if we can't get the system memory. Default: 50 MiB
* Have we hit the minimum allowed CPU idle time? Ignore if we can't get the CPU usage. Default: 15%
* Have we hit the maximum allowed I/O wait time? Ignore if we can't get the CPU usage. Default: 10%

These limits ensure that if we have way too many tasks which demand immediate execution we won't bring a server to its knees. The max number of concurrent tasks should be selected in a way that does not cause the memory to run out, or TCP/IP ports to be depleted. Anything over 50 is probably asking for trouble on a standard VPS.

Note that once this task is done it's possible to scale Panopticon a bit better (with the critical constraint being MySQL – using Redis would've been better, but we had to make a trade-off for shared server compatibility). One could launch more server instances of Panopticon which have the same configuration as the “main” server, and only run the Panopticon daemon.

For this reason we need to provide two things:
* A pair of events. One to be called when the daemon has been unable to run the Task Runner task for more than X consecutive seconds (configurable); this is the _scale up_ event. Another one to be called when the Task Runner found no tasks for X consecutive seconds (configurable). This is the _scale down_ event.
* Use the user_code architecture to plug user code into the aforementioned events.

This would allow for automatic scaling up and down for those users who actually do really need this.

**Comment by nikosdion** [comment]
With the introduction of the `--loop` argument to the task runner CLI command this is no longer relevant.
